src/common/config.h

```diff
@@ -136,6 +212,8 @@ struct md_config_t {
   // No metavariables will be returned (they will have already been expanded)
   int get_val(const char *key, char **buf, int len) const;
   int _get_val(const char *key, char **buf, int len) const;
+  config_value_t get_val(const char *key) const;
```

Comment:
It might be nice to just template this method on the expected value type -- that would eliminate the need for each caller to have to cast the variant and check for failure.